lib/validators/slides-validator.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Validates value of slides property
+ */
+
+import { Validator } from './validator';
+import { ParsedComponentsDefinition, ParsedComponentsDefinitionComponent, ParsedComponentsDefinitionProperty } from '../models';
+
+export class SlidesValidator implements Validator {
+
+    constructor(
+        private parsedDefinition: ParsedComponentsDefinition,
+    ) {
+    }
+
+    validate(
+        errorReporter: (errorMessage: string) => void,
+    ): boolean {
+        let valid = true;
+
+        // Find slides control properties and check for include and exclude
+        for (const parsedComponent of Object.values(this.parsedDefinition.components)) {
+            parsedComponent.properties.forEach((property) => {
+                valid = this.validateComponent(errorReporter, parsedComponent, property) && valid;
+            });
+        }
+
+        return valid;
+    }
+
+    /**
+     * Validates when slides property is found for component.
+     * Checks if restrictChildren is set and validates the include and exclude properties
+     * are valid for the slide component of doc-slideshow.
+     *
+     * @param errorReporter
+     * @param parsedComponent
+     * @param property
+     */
+    validateComponent(
+        errorReporter: (errorMessage: string) => void,
+        parsedComponent: ParsedComponentsDefinitionComponent,
+        property: ParsedComponentsDefinitionProperty,
+    ): boolean {
+        if (property.control.type !== 'slides') {
+            return true;
+        }
+
+        if (!parsedComponent.component.restrictChildren) {
+            errorReporter(`Component "${parsedComponent.component.name}" must have restrictChildren set to use the slides property`);
```

Comment:
Hmm, there is restrict-children-validator as well. Now the whole validator checks if restrictChildren is present for doc-slideshow directive and for slides control. Probably it can be combained somehow. But it could be like it is now too I think

Reply:
I will see if I can pick it up in between, there's some other small things I would like to refactor as well (in particular properties list being in two places). However will merge it now to progress on the slideshow properties.